### PR TITLE
fixing problems with skeleton tracks with missing points

### PR DESCRIPTION
**Pull request by Eldies**
### Motivation and context

fixing https://github.com/cvat-ai/cvat/issues/7783
also fixing related problem - some jobs from the issue can not be opened in ui due to missing shapes

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **New Features**
	- Enhanced data integrity in annotation processing by adding filtering steps before sorting.
	- New entries for "many jobs skeleton tracks with missing shapes" and "many jobs skeleton" added to the dataset, improving job tracking capabilities and complex job definitions.
	- Introduced a new static method for annotation creation, improving code modularity and reusability.

- **Bug Fixes**
	- Improved error handling in annotation functions to prevent runtime errors when accessing potentially undefined variables.

- **Tests**
	- Expanded the testing framework with new tests for annotation creation scenarios, ensuring comprehensive coverage.

### CODE REVIEW

**Comment by Bobronium** [comment]
Seems like SonarCloud Code Analysis is [failing](https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN,CONFIRMED&types=CODE_SMELL&pullRequest=8377&id=cvat-ai_cvat&open=AZGfcdstq2nxm6zF2sfN) because of the addition of `_make_annotations_for_task`. Now,  I see that it actually reduced complexity compared to previous implementation, however since it's technically a new function, SonarCloud feels entitled to complain. 

I suggest either refactoring this function further to meet the complexity criteria: `Refactor this function to reduce its Cognitive Complexity from 75 to the 15 allowed.`, or figuring out the way to suppress the warning. I couldn't quickly find it, but I imagine there has to be one.

Otherwise LGTM!

File: cvat/apps/dataset_manager/tests/assets/annotations.json [hunk 1]
```diff
@@ -416,6 +416,50 @@
...
+            "group": 0,
+            "source": "file",
+            "shapes": [],
+            "attributes": [],
+            "label_id": null
+          }
+        ],
+        "label_id": null
+      }
+    ]
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/tests/assets/annotations.json:461]: Probably, it makes sense to add a couple more tests:
- the same, but the skeleton shape with `outside` = `true`
- the same, but some of the points are present

**Comment by Eldies** [comment]
> I suggest either refactoring this function

refactored it

File: cvat/apps/dataset_manager/tests/assets/annotations.json (hunk 1 above)
Comment (Eldies) [reply cvat/apps/dataset_manager/tests/assets/annotations.json]: added tests

**Review by zhiltsov-max — APPROVED** [review]

**Review by Bobronium — APPROVED** [review]

**Pull request merged**